bigquery_etl/cli/query.py

```diff
@@ -0,0 +1,144 @@
+)
+
+
+def create_query_cli_group():
+    """Create the CLI group for the query command."""
+    group = click.Group(name="query", help="Creating/validating/deleting/... queries")
+    group.add_command(query_create_command)
+    return group
+
+
+@click.command(
+    name="create",
+    help=(
+        "Create a new query with name "
+        "<dataset>.<query_name>, for example: telemetry_derived.asn_aggregates"
+    ),
+)
+@click.argument("name")
+@click.option(
+    "--path",
+    "-p",
+    help="Path to directory in which query should be created",
+    default="sql/",
+)
+@click.option(
+    "--owner",
+    "-o",
+    help="Owner of the query (email address)",
+    default="example@mozilla.com",
+)
+@click.option(
+    "--init",
+    "-i",
+    help="Create an init.sql file to initialize the table",
+    default=False,
+)
+def query_create_command(name, path, owner, init):
+    """CLI command for creating a new query."""
+    if not os.path.isdir(path):
+        click.echo(f"Invalid path for adding new query: {path}", err=True)
+        sys.exit(1)
+
+    # create directory structure for query
+    try:
+        match = QUERY_NAME_RE.match(name)
+        name = match.group("name")
+        dataset = match.group("dataset")
+        version = match.group("ver") or "_v1"
+    except AttributeError:
+        click.echo(
+            "New queries must be named like:"
+            + " <dataset>.<table> or <dataset>.<table>_v[n]"
+        )
+        sys.exit(1)
+
+    derived_path = None
+    view_path = None
+
+    if dataset.endswith("_derived"):
+        # create a directory for the corresponding view
+        derived_path = Path(path) / dataset / (name + version)
+        os.makedirs(derived_path)
+
+        view_path = Path(path) / dataset.replace("_derived", "") / name
+        os.makedirs(view_path)
+    else:
+        # check if there is a corresponding derived dataset
+        if (Path(path) / (dataset + "_derived")).exists():
+            derived_path = Path(path) / (dataset + "_derived") / (name + version)
+            os.makedirs(derived_path)
+            view_path = Path(path) / dataset / name
+            os.makedirs(view_path)
+
+            dataset = dataset + "_derived"
+        else:
+            # some dataset that is not specified as _derived
+            # don't automatically create views
+            derived_path = Path(path) / dataset / (name + version)
+            os.makedirs(derived_path)
+
+    click.echo(f"Created query in {derived_path}")
+
+    if view_path:
+        click.echo(f"Created corresponding view in {view_path}")
+        view_file = view_path / "view.sql"
+        view_dataset = dataset.replace("_derived", "")
+        view_file.write_text(
+            (
+                "CREATE OR REPLACE VIEW\n"
+                f"  `moz-fx-data-shared-prod.{view_dataset}.{name}`\n"
+                "AS SELECT * FROM\n"
+                f"  `moz-fx-data-shared-prod.{dataset}.{name}{version}`\n"
+            )
```

Comment:
The default queries defined here should use `reformat`, so that `script/format_sql` doesn't need to be called after runnning this command:
```suggestion
            reformat(
                "CREATE OR REPLACE VIEW\n"
                f"  `moz-fx-data-shared-prod.{view_dataset}.{name}`\n"
                "AS SELECT * FROM\n"
                f"  `moz-fx-data-shared-prod.{dataset}.{name}{version}`"
            ) + "\n"
```

Reply:
Since this section is getting reformatted too, it might be nicer to use a block string:

```suggestion
           reformat(f"""
                CREATE OR REPLACE VIEW
                  `moz-fx-data-shared-prod.{view_dataset}.{name}`
                AS SELECT * FROM
                  `moz-fx-data-shared-prod.{dataset}.{name}{version}`
            """)
```